### DISCUSSION TITLE
ESP8266 (Wemos D1 ESP12) wifi -> GSM via wifi disturbance

### DISCUSSION THREAD

**Discussion by Fre19** [micropython/micropython#10117] [Using MicroPython] [answered]
I have a home application (all micropython) that works like a charm: ESP12 (Wemos D1) connected to a Pico, both 'very close' to my Wifi-router. The Pico is a logger for radiological data + a home intrusion security system. The system sends all info to a MQTT broker and to IFTTT for immediate alerts. So far so good. It is operational more than a year now.

Since many months I get response from mobile callers that somethimes I dissapear for a short time and that they almost don't hear me for a short while. I came to the conclusion, having excluded my iPhone as the cause, that very probably my home application is consuming a lot of wifi bandwidth, when both GSM and application are in the same wifi channel. I also assume that very strong Wifi signals are getting priority in my Wifi-router. I measured the wifi power from my ESP appl, this is at about -50dBm.

Question:
Is there any command in micropython to reduce the tx power from my home appl ? 
I tried things like wlan=network.WLAN(), wlan.config(txpower=-70) e.g. but apparently 'config' is not even available in micropython.

Meanwhile I'll continue testing this phenomena by switching my appl OFF during long phone calls and GSM call tests via 4G as well.
Many thx for your feedback

**Comment by peterhinch** [comment]
I'm very surprised to hear of WiFi affecting GSM. My understanding is that they use entirely different radio frequencies.

I've run a number of long term tests with MicroPython targets running MQTT. Most MQTT applications use minimal bandwidth. I never detected any interference with other WiFi applications, still less with GSM. Are you sure it's your app that's causing the dropouts and not something completely different?

**Comment by Fre19** [comment]
Thanks Peter for your feedback, much appreciated!

indeed this is a tricky problem which I face for many months now. I realy would like to solve it and I will not give up, The iPhone is renewed and all kinds of tests on my wifi-router are succesful. I even looked for a testing application that shows the real-time wifi channel load, but these things are pretty complex to setup.

When I call via my local wifi network, I do have that problem. E.g. calling via Whatsapp which is using wifi as a bridge, VoIP. Just only yesterday by coincidence I was calling close to my home system and the called person suddenly could not hear me. So I am pretty sure, there is a wifi jam on one of the wifi channels. That is why I would like to reduce Wifi power without succes so far.
I can confirm though that when I am close (1-2m) to my home application, the called person simply cannot hear me, when I am further away (8m) it takes a while but during long phone calls, there are certainly moments that drop outs occur.
Meanwhile:
My simple plan is to switch off the MQTT application when I get a call and see if I notice a difference which I expect. 

I'll keep you posted of any progress of the tests
Anny suggestions are verry much appreciated

**Reply by davefes** [reply]
I could imagine a GSM phone with poor 3rd harmonic rejection "might" affect WiFi  operation.  Keywords ... blocking and receiver over-load.  What frequency is the GSM phone on?  Can you try another GSM phone that uses a different provider?

**Comment by Fre19** [comment]
Thanks davefes, good point 
however as I understood these communication systems, frequencies (2.4GHz for wifi in my case and for smartphone GSM 850, 900, 1800, 1900 MHz) may overlap since both systems are using dedicated frequency channels which are mutual excusive. But: when GSM uses Wifi as a bridge, I would assume that GSM packets do use wifi channels. Furthermore the way frequency hopping is used to overcome blocking and overload is not verry transparent to put it mildly.

my plan of attack can be extended now and will be as follows:
1. Switching my micropython application off during a phone call to verify if comms improves (no weak gsm comms)
2. Try to find a way to reduce wifitxpower during phone calls with my iphone 8 and check if there is correlation
3. Test with another smartphone if the effects remain

I'll keep you'll posted of any progress of the tests
Anny suggestions are verry much appreciated

**Comment by Fre19** [comment]
I am still testing why I can't phone via wifi close to my ESP8266 application with micropython. I also noticed something strange: my application is in STA_IF mode, still the device presents itself as a Wifi-network 'MicroPython-dd268a'. I would not expect this in STA_IP mode, only in AP mode. Maybe my iPhone selects the strongest wifi network automatically and is fooled by the ESP8266 wifi?

Question:
Is it correct that in STA_IF the ESP device also presents itself as a wifi network?
Can it be switched off?

Thanks for your reply

**Reply by robert-hh** [reply]
The ESP8266 re-enables the previous WiFi state. So if AP has been enabled once, it will so at every boot. It can be disabled by  starting the AP mode and then calling active(False).

```
import network
wlan = network.WLAN(network.AP_IF) # create AP IF
wlan.active(False)       # deactivate the interface
```

**Comment by Fre19** [comment]
Thx robert-hh,
I will force AP False and try that out on wifi iPhone comms.

**Comment by Fre19** [comment, marked as answer]
**Solved**: GSM via wifi disturbance
I veryfied the comment of robert-hh and indeed, the esp8266 was in AP mode even when the script did NOT set the AP mode. I tested this further by changing the script and forcing AP mode and then back to STA, indeed the esp8266 still created a MicroPython-xxxx network.

In my script, I included #:
...
wlan.connect(wifi_ssid, wifi_password)
network.WLAN(network.AP_IF).active(False) # force Access Point OFF
...
The MicroPython-xxx wifi network disappeard and smartphone gsm comm via wifi is now flawless.
I checked with the smartphone app AirPort and found that my Home wifi network and MicroPython had about the same dB strength which support my assumption now that 'a smartphone can be fooled by the MicroPython network leading to bad phone connection via wifi'.

I suggest therefore to INCLUDE the # force Access Point OFF instruction in STA mode.

Thx for your suport in this (long lasting) issue

**Comment by hzy-zg** [comment]
已收到来信，我会尽快处理。
祝每天好心情！
